### PR TITLE
feat(content-section): web-component version

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)

#3736 

### Description

Created the internal component to use within the future Web Components version of `CardSection`.

### Changelog

**New**

- Introducing Content Section for Web Components.

### CODE REVIEW

File: packages/web-components/src/components/content-section/content-section.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  /**
+   * The Content Section heading.
+   */
+  @property()
+  heading = '';
+
+  @property({ reflect: true })
+  theme = CONTENT_SECTION_THEME.WHITE;
```
Comment (asudoh) [inline packages/web-components/src/components/content-section/content-section.ts:90]: We are going away with setting `theme` property to components, given we are using CSS custom properties to set the theme.

File: packages/web-components/src/components/content-section/content-section.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    `;
+  }
+
+  /**
+   * Renders children
+   */
+  protected __renderChildren() {
+    const { children } = this;
+    return html`
+      <slot name="children">${children}</slot>
```
Comment (asudoh) [inline packages/web-components/src/components/content-section/content-section.ts:79]: Web Components has no notion like React's `children` property. The closest to React `children` property is the default `<slot>`:

```suggestion
      <slot></slot>
```

File: packages/web-components/src/components/content-section/content-section.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+   */
+  @property()
+  heading = '';
+
+  @property({ reflect: true })
+  theme = CONTENT_SECTION_THEME.WHITE;
+
+  @property()
+  customClassName = '';
+  // data-autoid={otherProps.autoid ? otherProps.autoid : "${prefix}--content-section"}
```
Comment (asudoh) [inline packages/web-components/src/components/content-section/content-section.ts:94]: We use `StableSelectorMixin` to set `data-autoid`:
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/footer/footer.ts#L13
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/footer/footer.ts#L43
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/src/components/footer/footer.ts#L70-L72

File: packages/web-components/src/components/content-section/content-section.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  /**
+   * Returns a class-name based on the type parameter type
+   */
+  protected _getTypeClass() {
+    return classMap({
+      [`${this.customClassName}`]: this.customClassName,
+      [`${prefix}--content-section--${this.theme}`]: true,
+      [`${prefix}--content-section`]: true,
+    });
+  }
```
Comment (asudoh) [inline packages/web-components/src/components/content-section/content-section.ts:61]: Seems that this method is not used.

**Review by asudoh — COMMENTED** [review]
Nice progress @IgnacioBecerra!

**Comment by IgnacioBecerra** [comment]
@asudoh Thank you for your comments, the changes are now present in the current commit.

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @IgnacioBecerra for the update!

**Pull request merged**